internal/kafka/command_topic_produce.go

```diff
@@ -74,6 +79,22 @@ func (c *command) newProduceCommand() *cobra.Command {
 }
 
 func (c *command) produce(cmd *cobra.Command, args []string) error {
+	if c.Context == nil || c.Context.State == nil {
+		cobra.CheckErr(cmd.MarkFlagRequired("kafka-bootstrap"))
```

Comment:
This function can only be called when the command is initialized, right?

Reply:
If it's not logged in, even though the cmd is initialized, the context is still nil and empty...

Reply:
Right, but this function is called during runtime, at which point the command has already been parsed to determine if required flags are present or missing.